### PR TITLE
Fixed #51 Cannot use MakeNote and noted_under in the same Actions lis…

### LINKED ISSUE #51: Cannot use `MakeNote` and `noted_under` in the same Actions list
Say we have this:

```python
the_actor.attempts_to(
    MakeNote.of_the(Text.of_the(EXAMPLE_FIELD)).as_("notable note"),
    Enter.the_text(noted_under("notable note")).into_the(EXAMPLE_INPUT),
)
```

We get a `KeyError` from that because the Director's notebook is empty. The `MakeNote` action hasn't been performed yet!!

How to solve this quandary? Is there some soup of dunder-methods available that can delay actually getting the value from the Director's notebook until the Action's `perform_as` asks for it? Figure it out!

**P.S.:** There is a workaround, but i hate it:

```python
the_actor.attempts_to(
    MakeNote.of_the(Text.of_the(EXAMPLE_FIELD)).as_("notable note")
)
# just split them up 
the_actor.attempts_to(
    Enter.the_text(noted_under("notable note")).into_the(EXAMPLE_INPUT),
)

### PR DESCRIPTION
**Pull request by WandyYing**
Fixed #51

### CODE REVIEW

File: screenpy/director.py [hunk 1]
```diff
@@ -15,11 +15,8 @@ class Director:
 
     _instance = None
 
-    def __new__(cls) -> "Director":
-        if cls._instance is None:
-            cls._instance = super().__new__(cls)
-            cls._instance.notebook = {}
-        return cls._instance
+    def __init__(self):
+        self.notebook = {}
```
Comment (perrygoy) [inline screenpy/director.py:19]: This change actually makes this class *not* a singleton anymore. Calling `SingletonDirector()` will make new ones.

**Review by perrygoy — CHANGES_REQUESTED** [review]
I think i might not be fully understanding this change. Can you explain why this approach is better?

**Comment by WandyYing** [comment]
Technically, Python modules ARE singletons. This approach is simple and not lazily instantianted.  FYI.  [Using module as a singleton in Python - is that ok?
](https://stackoverflow.com/questions/52929033/using-module-as-a-singleton-in-python-is-that-ok)

**Comment by perrygoy** [comment]
> Technically, Python modules are singletons.

Yes, and thank you for the link, but i'm not understanding the reasoning for switching between two different ways to accomplish a singleton. Your comment on #51 said that this PR fixes the issue with `Director`—what was the issue?

**Comment by WandyYing** [comment]
The request is try to fix it #51. It can make the following methods work.
```
the_actor.attempts_to(
    MakeNote.of_the(Text.of_the(EXAMPLE_FIELD)).as_("notable note"),
    Enter.the_text(noted_under("notable note")).into_the(EXAMPLE_INPUT),
)
```
I think the reason for this problem is that there is no pre instantiation. So pre instantiation might solve the problem, so I tried.

But, maybe, it doesn't have to change the way it's implemented. I might close the request if the following is possible.

```
class __Director:
    """The single Director of the screenplay.

    The Director keeps track of information for the Actors. This information
    can be retrieved using a |direction|.
    """

    notebook: dict

    _instance = None

    def __new__(cls) -> "Director":
        if cls._instance is None:
            cls._instance = super().__new__(cls)
            cls._instance.notebook = {}
        return cls._instance

    ..... 

Director = __Director()
```
I will verify this approach later.

**Comment by perrygoy** [comment]
Aaah, i see.

The problem is not the pre-instantiation. What is causing #51 is that `MakeNote` hasn't yet been _performed_ when `noted_under` tries to retrieve the note. 

`noted_under` tries to retrieve the note when all the actions are being passed in to `attempts_to`, while `MakeNote` won't actually give the note to the `Director` until `MakeNote` is actually performed _inside_ of `attempts_to`. 

Does that help explain it?

**Comment by WandyYing** [comment]
> `noted_under` tries to retrieve the note when all the actions are being passed in to `attempts_to`, while `MakeNote` won't actually give the note to the `Director` until `MakeNote` is actually performed _inside_ of `attempts_to`.

Yes, I agree with you. And when I checked again, I found that my test case design was wrong, which led to a false pass. I mistook the method for working. And please close the RP. lol

**Pull request closed without merging**